test/css_mirroring/transformer_test.dart

```diff
@@ -0,0 +1,231 @@
+    '''
+  }, {
+    'a|foo2_unmatched_css_url.css': r'''
+
+
+    '''
+  });
+
+  /// Changes rules contaning only direction dependent declarations to
+  /// direction dependent rules without any rule for common values
+  testPhases('drops orientation-neutral rules when they are empty', phases, {
+    'a|foo2_unmatched_css_url.css': r'''
+          absent-element {
+            float: left;
+            margin-left: 100px;
+        }
+          .usedclass {
+            padding: right;
+            text-align: left;
+        }
+        '''
+  }, {
+    'a|foo2_unmatched_css_url.css': r'''
+
+          :host-context([dir="ltr"]) absent-element {
+            float: left;
+            margin-left: 100px;
+        }
+          :host-context([dir="ltr"]) .usedclass {
+            padding: right;
+            text-align: left;
+        }
+
+          :host-context([dir="rtl"]) absent-element {
+            float: right;
+            margin-right: 100px;
+        }
+          :host-context([dir="rtl"]) .usedclass {
+            padding: left;
+            text-align: right;
+        }
+        '''
+  });
+
+  testPhases('leaves orientation-neutral rules unchanged', phases, {
+    'a|foo2_unmatched_css_url.css': r'''
+          absent-element {
+            color: blue;
+        }
+          .usedclass {
+            background-size: 16px 16px;
+        }
+        '''
+  }, {
+    'a|foo2_unmatched_css_url.css': r'''
+          absent-element {
+            color: blue;
+        }
+          .usedclass {
+            background-size: 16px 16px;
+        }
+
+
+        '''
+  });
+
+  testPhases(
+      'splits the exotic rules with language dependent declaration to common rule and language dependent rules',
+      phases, {
+    'a|foo2_unmatched_css_url.css': r'''
+        li + li {
+          color: blue;
+          float: right;
+        }
+        a ~ a {
+          color: purple;
+          margin-left: 3em;
+        }
+        li > a {
+          color: orange;
+          margin-left: 4em; width: 10px;
+        }
+        '''
+  }, {
+    'a|foo2_unmatched_css_url.css': r'''
+        li + li {
+          color: blue;
+          }
+        a ~ a {
+          color: purple;
+          }
+        li > a {
+          color: orange;
+          width: 10px;
+        }
+
+        :host-context([dir="ltr"]) li + li {
+          float: right;
+        }
+        :host-context([dir="ltr"]) a ~ a {
+          margin-left: 3em;
+        }
+        :host-context([dir="ltr"]) li > a {
+          margin-left: 4em; }
+
+        :host-context([dir="rtl"]) li + li {
+          float: left;
+        }
+        :host-context([dir="rtl"]) a ~ a {
+          margin-right: 3em;
+        }
+        :host-context([dir="rtl"]) li > a {
+          margin-right: 4em; }
+        '''
+  });
+
+  /// Keeps direction independent directives like
+  /// @charset, @import, @font-face, @namespace same.
+  testPhases('keeps direction independent directives same', phases, {
```

Comment:
So if it's a same, use the same string (extract to local var) for both contents, something like:

```dart
var directionIndependentDirectives = {
  'charset': '''@charset "...";''',
  'font-face': '''
    @font-face {
      ...
    }
  ''',
  ...
};
directionIndependentDirectives.forEach((directive, css) {
  testPhases('keeps $directive directive untouched', phases, {
    'a|foo2_unmatched_css_url.css': css
  }, { {
    'a|foo2_unmatched_css_url.css': css
  });
});
```

This will allow your tests to fail independently (in case one of the directives regresses)